grails-test-suite-uber/src/test/groovy/grails/compiler/GrailsCompileStaticCompilationErrorsSpec.groovy

```diff
@@ -188,6 +188,42 @@ class SomeClass implements grails.validation.Validateable {
         c
     }
 
+    @Issue('GRAILS-12101')
+    void 'Test compiling mapping with inner class'() {
+        given:
+        def gcl = new GroovyClassLoader()
+
+        when: 'a class marked with @GrailsCompileStatic invokes dynamic finders on a non-domain class inside of a method marked with TypeCheckingMode.SKIP'
```

Comment:
This description on the `when` block here and in line 173 is I guess a copy & paste error from line 144 and should be removed or corrected.